src/app/public/src/modules/nav/nav.service.ts

```diff
@@ -45,19 +42,27 @@ export class StacheNavService {
     let activeRoutes = [{
       path: rootPath,
       segments: [rootPath],
-      children: this.orderRoutes(activeChildRoutes, rootPath)
+      children: this.assignChildren(activeChildRoutes, rootPath)
     }];
 
     this.activeRoutes = this.formatRoutes(activeRoutes);
 
     return this.clone(this.activeRoutes) as StacheNavLink[];
   }
 
+  public getActiveUrl(): string {
+    return this.router.url.split('#')[0];
+  }
+
   public clearActiveRoutes() {
     this.activeRoutes = undefined;
   }
 
-  private orderRoutes(routes: any[], parentPath: string): any[] {
+  private clone(thing: any): any {
+    return JSON.parse(JSON.stringify(thing));
+  }
+
+  private assignChildren(routes: any[], parentPath: string): any[] {
     const ordered: any[] = [];
```

Comment:
Can you think of a better name than `ordered`, since the name of the function has changed? Maybe `branch`?